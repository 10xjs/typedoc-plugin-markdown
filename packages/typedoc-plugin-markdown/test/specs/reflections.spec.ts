import * as Handlebars from 'handlebars';

import { TestApp } from '../test-app';

describe(`Reflections:`, () => {
  let testApp: TestApp;

  let reflectionTemplate: Handlebars.TemplateDelegate;

  beforeAll(() => {
    testApp = new TestApp(['reflections.ts']);
    testApp.bootstrap();
    TestApp.stubPartials([
      'header',
      'index',
      'comment',
      'member.signature',
      'members',
    ]);
    TestApp.stubHelpers(['breadcrumbs', 'hierarchy']);

    reflectionTemplate = TestApp.getTemplate('reflection');
  });

  beforeEach(() => {
    Handlebars.registerHelper('ifShowBreadcrumbs', function (options) {
      return options.fn(this);
    });
  });

  test(`should compile module without breadcrumbs`, () => {
    Handlebars.registerHelper('ifShowBreadcrumbs', function (options) {
      return options.inverse(this);
    });
    expect(
      TestApp.compileTemplate(reflectionTemplate, {
        model: testApp.findModule('reflections'),
      }),
    ).toMatchSnapshot();
  });

  test(`should compile reflection with type params`, () => {
    expect(
      TestApp.compileTemplate(reflectionTemplate, {
        model: testApp.findReflection('ReflectionWithTypeParams'),
      }),
    ).toMatchSnapshot();
  });

  test(`should compile a callable reflection`, () => {
    expect(
      TestApp.compileTemplate(reflectionTemplate, {
        model: testApp.findReflection('CallableReflection'),
      }),
    ).toMatchSnapshot();
  });

  test(`should compile an indexable reflection`, () => {
    expect(
      TestApp.compileTemplate(reflectionTemplate, {
        model: testApp.findReflection('IndexableReflection'),
      }),
    ).toMatchSnapshot();
  });
});
