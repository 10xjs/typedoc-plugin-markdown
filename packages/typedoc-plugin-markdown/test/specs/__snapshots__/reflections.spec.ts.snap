// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reflections: should compile a callable reflection 1`] = `
"## Hierarchy

[helper: hierarchy]

## Callable

[partial: member.signature]

[partial: index]

[partial: members]
"
`;

exports[`Reflections: should compile an indexable reflection 1`] = `
"## Hierarchy

[helper: hierarchy]

## Indexable

▪ \\\\[index: *number*\\\\]: *string*

[partial: comment]

[partial: index]

[partial: members]
"
`;

exports[`Reflections: should compile default layout with breadcrumbs 1`] = `
"[helper: breadcrumbs]

# Class: ReflectionClass

CONTENTS
"
`;

exports[`Reflections: should compile default layout with type params in title 1`] = `
"[helper: breadcrumbs]

# Interface: ReflectionWithTypeParams ‹**T, V**›

CONTENTS
"
`;

exports[`Reflections: should compile default layout without breadcrumbs 1`] = `
"# Class: ReflectionClass

CONTENTS
"
`;

exports[`Reflections: should compile reflection with comment and without index 1`] = `
"[partial: comment]

## Hierarchy

[helper: hierarchy]

[partial: members]
"
`;

exports[`Reflections: should compile reflection with type params 1`] = `
"## Type parameters

* T
* V

## Hierarchy

[helper: hierarchy]

[partial: index]

[partial: members]
"
`;
